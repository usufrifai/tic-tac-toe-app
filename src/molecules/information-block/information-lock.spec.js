import React from 'react';
import InformationlBock from './index';
import { Button } from 'react-bootstrap';
import { create, act } from 'react-test-renderer';

describe('InformationlBock', () => {
    it('InformationlBock should be rendered', () => {
        const component = create(
            <InformationlBock 
            winnerMessage="Game Draw"
            users={{firstPlayer: 'Yousuf', secondPlayer:'Rifay'}} 
            resetGame={jest.fn()}
            newGame={jest.fn()}
            />
          );
          
          const instance = component.root;
          const button = instance.findAllByType(Button);
          act(() => {
                button[0].props.onClick();
                button[1].props.onClick();
          });
        
          let tree = component.toJSON();
          expect(tree).toMatchSnapshot();
    });
  });