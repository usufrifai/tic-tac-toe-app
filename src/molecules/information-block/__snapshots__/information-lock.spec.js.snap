// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationlBock InformationlBock should be rendered 1`] = `
<div
  className="information-container"
>
  <div
    className="heading"
    style={
      Object {
        "width": undefined,
      }
    }
  >
    <h3>
      Information
    </h3>
  </div>
  <div
    className="list-group"
    onKeyDown={[Function]}
  >
    <div
      className="list-group-item list-group-item-warning"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      Player One : 
      Yousuf
    </div>
    <div
      className="list-group-item"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      Player Two : 
      Rifay
    </div>
  </div>
  <img
    alt="Winner"
    className="trophy"
    src="winner.gif"
  />
  ;
  <h4>
    Game Draw
  </h4>
  <button
    className="btn btn-warning"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Start New Game
  </button>
   
  <button
    className="btn btn-secondary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Play Again
  </button>
</div>
`;
